### PR TITLE
add service monitor to component reconciler

### PR DESCRIPTION
**Pull request by ruanxin**
**Description**
- enable service monitor for each component reconciler
- set PeerAuthentication Policies to PERMISSIVE mode allow both mTLS & plaintext traffic
- set Authorization Policy to allow access metrics endpoint in istio service mesh

**Related issue(s)**
https://github.com/kyma-incubator/reconciler/issues/907

### CODE REVIEW

File: resources/kcp/charts/component-reconcilers/templates/policy.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  labels:
+    component: {{ $component }}
+  name: {{ $component }}-reconciler
+  namespace: {{ $.Release.Namespace }}
+spec:
+  selector:
+    matchLabels:
+      component: {{ $component }}
+  mtls:
+    mode: PERMISSIVE
```
Comment (adityabhatia) [inline resources/kcp/charts/component-reconcilers/templates/policy.yaml:15]: Would it make sense to restrict all workloads with the matched label on a specific (http) port?
```suggestion
  mtls:
    mode: STRICT
  portLevelMtls:
    8080:
      mode: PERMISSIVE
```

Comment (ruanxin) [reply resources/kcp/charts/component-reconcilers/templates/policy.yaml:15]: cool, don't know this feature, that's totally valid point.

**Review by adityabhatia — DISMISSED** [review]

**Review by adityabhatia — APPROVED** [review]

**Pull request merged**